src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -2081,10 +2081,11 @@ private void InsertWorkspace(WorkspaceModel ws, ExtraWorkspaceViewInfo viewInfo
         {
             var nodes = ws.Nodes;
             var connectors = ws.Connectors;
+            var notes = viewInfo?.Annotations;
 
             DynamoSelection.Instance.ClearSelection();
 
-            if (nodes == null || !nodes.Any() || NodesAlreadyLoaded(nodes))
+            if ((nodes == null && notes == null) || (!nodes.Any() && !notes.Any()) || (NodesAlreadyLoaded(nodes) || NotesAlreadyLoaded(notes)))
```

Comment:
don't you want Empty here instead of !Any?

Reply:
Totally, but Workspace.Nodes is IEnumerable, it doesn't have .Empty(). I actually wouldn't have thought of using .Empty() in any case, so thanks for the prompt! I will leave it as is, if that's ok.